### ISSUE TITLE
JIT alloc.h missing functionality to warn against using new and delete.

### ISSUE THREAD

**Issue by jashook** [dotnet/coreclr#1913]
Overloading the new and delete operator in alloc.h was removed to fix the build. See: 347ab205ca8a48ef445101b7a2e1499a9f59c596.

However, there should be some type of way to warn against using new or delete.

**Comment by RussKeldorph** [comment]
@pgavlin @russellhadley If I recall, this was done due to a conflict with the larger CLR.  Can this PR be undone with @pgavlin's pending changes?  If so, could you take it?

**Comment by russellhadley** [comment]
Sure.  @pgavlin can you take a look?

**Comment by jashook** [comment]
I believe this was removed because the toolset treated static operator new as an error after vs 2015 update 1. It should be independent of the clr.

See the discussion in the pr here for more details https://github.com/dotnet/coreclr/pull/1903.

**Comment by RussKeldorph** [comment]
@jashook Right, but I thought the reason the desired workaround (remove static) didn't work was because of conflicting definitions in CoreCLR.  In theory, if these are the only overrides inside the separated JIT DLL/.so, there would be no conflict.  I'm assuming JIT-internal conflicts either don't exist or can be removed at our whim.

If I'm off base, nevermind.  Also, if this isn't trivial to fix, we can move it to future.  I don't think it would block the release.

**Comment by jashook** [comment]
No I believe you are correct, I forgot this aspect of the issue.
